plateau/core/common_metadata.py

```diff
@@ -78,6 +78,33 @@ def _schema_compat(self):
                     if index_level_ix >= 0:
                         schema = schema.remove(index_level_ix)
 
+            for cmd in pandas_metadata["columns"]:
+                name = cmd.get("name")
+                if name is None:
+                    continue
+
+                field_name = cmd["field_name"]
+                field_idx = schema.get_field_index(field_name)
+                if field_idx < 0:
+                    continue
+                field = schema[field_idx]
+                if (
+                    pa.types.is_string(field.type)
+                    and cmd["pandas_type"] == "unicode"
+                    and cmd["numpy_type"] == "object"
+                ):
+                    schema = schema.remove(field_idx)
+                    new_field = pa.field(
+                        field.name,
+                        pa.large_string(),
+                        field.nullable,
+                        field.metadata,
+                    )
+                    schema = schema.insert(field_idx, new_field)
+                    cmd["pandas_type"] = "object"
+                    cmd["numpy_type"] = "str"
+                    cmd["metadata"] = None
```

Comment:
this turned out to be a bit more nasty than I was hoping. We'll need this compat code to ensure backwards compatibility with datasets written by pandas 2 when reading in pandas 3. Also more generally to ensure compatibility between string and object dtypes in pandas 3 (however unlikely that would be).